service/src/main/java/bio/terra/catalog/datarepo/DatarepoException.java

```diff
@@ -1,7 +1,15 @@
 package bio.terra.catalog.datarepo;
 
-public class DatarepoException extends RuntimeException {
-  public DatarepoException(String message, Throwable cause) {
-    super(message, cause);
+import bio.terra.common.exception.ErrorReportException;
+import bio.terra.datarepo.client.ApiException;
+import org.springframework.http.HttpStatus;
+
+public class DatarepoException extends ErrorReportException {
+  public DatarepoException(String message, ApiException e) {
+    super(message, e.getCause(), null, HttpStatus.valueOf(e.getCode()));
```

Comment:
I'm pretty sure you want `HttpStatus.resolve()` not `valueOf()`, since `valueOf()` will throw if it can't map the status code, and we wouldn't want to throw in a constructor for an exception.
```suggestion
    super(message, e.getCause(), null, HttpStatus.resolve(e.getCode()));
```

Also here again I would use `e` not `e.getCause()`, is there a reason to not use `e`?
```suggestion
    super(message, e, null, HttpStatus.valueOf(e.getCode()));
```

Reply:
No. I assumed typing-wise it wouldn't work and it does. Just some bad assumptions based off the existing code.